csharp/src/Ice/RemoteException.cs

```diff
@@ -133,4 +149,74 @@ public static class RemoteExceptionExtensions
         /// <returns>A SlicedData value that provides the list of sliced-off slices.</returns>
         public static SlicedData? GetSlicedData(this RemoteException ex) => ex.SlicedData;
     }
+
+    public partial class ObjectNotExistException
+    {
+        /// <inheritdoc/>
+        protected override string? DefaultMessage
+        {
+            get
+            {
+                if (Origin is RemoteExceptionOrigin origin)
+                {
+                    return $@"could not find servant for Ice object `{origin.Identity}'" +
+                        (origin.Facet.Length > 0 ? $" with facet `{origin.Facet}'" : "") +
+                        $" while attempting to dispatch operation `{origin.Operation}'";
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+    }
+
+    public partial class OperationNotExistException
+    {
+        /// <inheritdoc/>
+        protected override string? DefaultMessage
+        {
+            get
+            {
+                if (Origin is RemoteExceptionOrigin origin)
+                {
+                    return $"could not find operation `{origin.Operation}' for Ice object `{origin.Identity}'" +
+                        (origin.Facet.Length > 0 ? $" with facet `{origin.Facet}'" : "");
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+    }
+
+    public partial class UnhandledException : RemoteException
+    {
+        /// <summary>Constructs a new exception.</summary>
+        /// <param name="current">Holds decoded header data and other information about the current request.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
+        public UnhandledException(Current current, Exception innerException)
```

Comment:
I thought we could make it simply `UnhandledException(Exception innerException)` and fill-in the origin during marshaling like the other exceptions. 

Also use the DefaultMessage override to generate the message using Origin and innerException (when innerException is not null). When this exception is unmarshaled, the message/Message is set in its ctor and DefaultMessage is not used.